src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+
+    Herbivore.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+
+    Carnivore.alive.push(this);
+  }
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
+      animal.health -= 50;
+
+      if (animal.health === 0) {
+        const index = Animal.alive.indexOf(animal);
+
+        Animal.alive.splice(index, 1);
+      }
```

Comment:
[CODE PERFOMANCE] - Use the `filter` method to remove dead animals from the `Animal.alive` array to avoid nesting loops, as specified in the checklist. You can replace the `if` statement and the subsequent `indexOf` and `splice` operations with a single `Animal.alive = Animal.alive.filter(animal => animal.health > 0);`